baybe/surrogates/base.py

```diff
@@ -85,7 +89,7 @@ def to_botorch(self) -> Model:
 
 
 @define
-class Surrogate(ABC, SurrogateProtocol, SerialMixin):
+class Surrogate(ABC, SurrogateProtocol, SerialMixin, Generic[_ModelContext]):
```

Comment:
Why is this necessary? It feels weird that the surrogate inherits from what feels like a bare necessity that is required to make mypy happy.

Reply:
Not that this is **not** inheritance, it's purely a typing thing and the syntax required to declare generic classes in python until python 3.12 (in newer versions, this looks different). See:
https://mypy.readthedocs.io/en/stable/generics.html#defining-generic-classes

It lets us relate the output of `_get_model_context` to the input of `_fit` and thus allows the type checker to do its job properly. 

If more questions / still unhappy, let me know. Otherwise, resolve

Reply:
Ok, but why is it generic here? Is this simply because `_ModelContext` can basically be `Any`, and this is just the proper way to encode that?

Reply:
Makes sense :)